File: go/internal/lexer/token.go
package lexer

// TokenType es el tipo de un token de Jqmas.
type TokenType string

const (
// Literales
NUMBER TokenType = "NUMBER"
STRING TokenType = "STRING"

// Identificadores y palabras clave
PRES TokenType = "PRES"
HAZ  TokenType = "HAZ"
NOT  TokenType = "NOT"
IDENTIFIER TokenType = "IDENTIFIER"

// Operadores
PLUS   TokenType = "PLUS"
MINUS  TokenType = "MINUS"
STAR   TokenType = "STAR"
SLASH  TokenType = "SLASH"
EQUALS TokenType = "EQUALS"

// Delimitadores
LPAREN   TokenType = "LPAREN"
RPAREN   TokenType = "RPAREN"
LBRACKET TokenType = "LBRACKET"
RBRACKET TokenType = "RBRACKET"
COMMA    TokenType = "COMMA"

// Especiales
EOF TokenType = "EOF"
)

// Token representa un token del lenguaje Jqmas.
type Token struct {
Type  TokenType   `json:"type"`
Value interface{} `json:"value"`
Line  int         `json:"line"`
Col   int         `json:"col"`
}

File: src/jqmas/lexer/token.py
"""Tipos de token del lenguaje Jqmas."""

from dataclasses import dataclass
from enum import Enum, auto


class TokenType(Enum):
    """Tipos de token posibles en Jqmas."""

    # Literales
    NUMBER = auto()
    STRING = auto()

    # Identificadores y palabras clave
    PRES = auto()
    NOT = auto()
    IDENTIFIER = auto()

    # Operadores
    PLUS = auto()
    MINUS = auto()
    STAR = auto()
    SLASH = auto()

    # Delimitadores
    LPAREN = auto()
    RPAREN = auto()
    LBRACKET = auto()
    RBRACKET = auto()
    COMMA = auto()

    # Especiales
    EOF = auto()


@dataclass
class Token:
    """Un token del lenguaje Jqmas."""

    type: TokenType
    value: object
    line: int
    column: int

    def __repr__(self) -> str:
        return f"Token({self.type.name}, {self.value!r}, {self.line}:{self.column})"
